orderbook/src/database/orders.rs

```diff
@@ -321,6 +323,26 @@ impl OrderStoring for Postgres {
             .await
     }
 
+    async fn orders_for_tx(&self, tx_hash: &H256) -> Result<Vec<Order>> {
+        #[rustfmt::skip]
+        const QUERY: &str = concatcp!(
+            "SELECT ", ORDERS_SELECT,
+            "FROM ", ORDERS_FROM,
+            "JOIN trades t \
+                ON t.order_uid = o.uid \
+             JOIN settlements s \
+                ON s.block_number = t.block_number \
```

Comment:
Should also join on log index

Reply:
I really don't think we should be joining on log index. Each settlement and trade event have their own unique log index.

Reply:
We need the settlements to get the TxHash (it is not part of the trades table).

Reply:
Forgot the tx hash is in settlements not trades. 
As for the join you're right too that log index would be wrong. My confusions stems from settlement having as the primary key (block, long_index) so I thought we were not uniquely identifying settlements. Well, we kind of still are. I think in other code we have this "find the next larger log index in the same block" logic to get the settlemen. Not sure how feasible that is for this query.

Reply:
I'm just glad I was able to help clear things up here.